source/simple_sub_pub_demo.c

```diff
@@ -244,12 +271,28 @@ static void prvSubscribeCommandCallback( void * pxCommandContext,
                                          MQTTAgentReturnInfo_t * pxReturnInfo )
 {
     CommandContext_t * pxApplicationDefinedContext = ( CommandContext_t * ) pxCommandContext;
+    const char * pcTopicBuffer = ( const char * ) pxApplicationDefinedContext->pArgs;
 
     /* Store the result in the application defined context so the task that
      * initiated the subscribe can check the operation's status.  Also send the
      * status as the notification value.  These things are just done for
      * demonstration purposes. */
     pxApplicationDefinedContext->xReturnStatus = pxReturnInfo->returnCode;
+
+    /* Check if the subscribe operation is a success. Only one topic is
+     * subscribed by this demo. */
+    if( ( pxReturnInfo->returnCode == MQTTSuccess ) &&
+        ( pxReturnInfo->pSubackCodes[ 0 ] != MQTTSubAckFailure ) )
+    {
+        /* Add subscription so that incoming publishes are routed to the application
+         * callback. */
+        addSubscription( xGlobalSubscriptionList,
+                         pcTopicBuffer,
+                         ( uint16_t ) strlen( pcTopicBuffer ),
+                         prvIncomingPublishCallback,
+                         pxApplicationDefinedContext );
```

Comment:
This doesn't get used by the callback, so it should be NULL. Also, the members of the application defined context aren't guaranteed to stay in scope after this function completes, so doing this may make the demo confusing
```suggestion
                         NULL );
```